docs/developer/contributing.md

```diff
@@ -0,0 +1,193 @@
+
+In order to start developing a new feature or refactoring, a new branch should be created with one of the following
+names:
+
+-   `hardening/<taskName>`
+-   `task/<taskName>`
+-   `feature/<taskName>`
+
+depending on the kind of work.
+
+This branch must be created from the current version of the `master` branch. Once the new functionality has been
+completed, a Pull Request will be created from the feature branch to `master`. Remember to check both the linters and
+the tests before creating the Pull Request.
+
+Bugfixes work the same way as other tasks, with the exception of the branch name, that should be called `bug/<bugName>`.
+
+In order to contribute to the repository, these same scheme should be replicated in the forked repositories, so the new
+features or fixes should all come from the current version of `master` and end up in `master` again.
+
+All the `task/*` and `bug/*` branches are temporary, and should be removed once they have been merged.
+
+There is another set of branches called `release/<versionNumber>`, one for each version of the product. This branches
+point to each of the released versions of the project, they are permanent and they are created with each release.
+
+## Change log
+
+The project contains a version changelog, called CHANGES_NEXT_RELEASE, that can be found in the root of the project.
+Whenever a new feature or bug fix is going to be merged with `master`, a new entry should be added to this changelog.
+The new entry should contain the reference number of the issue it is solving (if any).
+
+When a new version is released, the change log is frozen, and CHANGES_NEXT_RELEASE remains fixed in the last commit of
+that version. The CHANGES_NEXT_RELEASE is flushed (i.e. all its content removed) when preparing for the next release
+developing cycle (i.e. during the task described in bullet 6 in next subsection). The contents of this file are also
+copied to the change log section in the `.spec` file for the component.
+
+## Releasing
+
+The process of making a release consists of the following steps:
+
+1. Create a new task branch changing the development version number in the package.json (with a sufix `-next`), to the
+   new target version (without any sufix), and PR into `master`. Also, in this task, the contents of the Change log are
+   copied to the RPM spec.
+2. Create a tag from the last version of `master` named with the version number and push it to the repository.
+3. Create the release in Github, from the created tag. In the description, add the contents of the Change log.
+4. Create a release branch from the last version of `master` named with the version number.
+5. Create a new task for preparing the next release, adding the sufix `-next` to the current version number (to signal
+   this as the development version), and flush the contents of the CHANGES_NEXT_RELEASE file.
+6. Upload the new library version to the npm repository using `npm publish` command from the release branch.
```

Comment:
```suggestion
```

`npm publish` is not applicable to Perseo-FE. It is only applicable to https://github.com/telefonicaid/iotagent-node-lib (if we have this bullet in any other project appart from it, it should be removed)

Reply:
Fixed in [c6ccdd0](https://github.com/telefonicaid/perseo-fe/pull/632/commits/c6ccdd0912de0a254500b1e288c7a56e8b08f3c5)